disnake/types/user.py

```diff
@@ -2,13 +2,16 @@
 
 from typing import Literal, Optional, TypedDict
 
+from typing_extensions import NotRequired
+
 from .snowflake import Snowflake
 
 
 class PartialUser(TypedDict):
     id: Snowflake
     username: str
-    discriminator: str
+    discriminator: str  # may be removed in future API versions
```

Comment:
I thought about it and looked into this more, and decided this isn't worth the hassle (and performance hit, minor but on a hot path) given reassurance that this won't break:
![image](https://github.com/DisnakeDev/disnake/assets/8530778/bd3c475f-2b9a-47eb-8d7b-aa467b6ab0e8)
This would be a really breaking change for the entire Discord ecosystem anyway, if anything I anticipate this happening in v11.